src/newgrp.c

```diff
@@ -451,120 +447,57 @@ int main (int argc, char **argv)
 	}
 	name = pwd->pw_name;
 
-	/*
-	 * Parse the command line. There are two accepted flags. The first
-	 * is "-", which for newgrp means to re-create the entire
-	 * environment as though a login had been performed, and "-c", which
-	 * for sg causes a command string to be executed.
-	 *
-	 * The next argument, if present, must be the new group name. Any
-	 * remaining arguments will be used to execute a command
-	 * as the named group. If the group name isn't present, I just use
-	 * the login group ID of the current user.
-	 *
-	 * The valid syntax are
-	 *      newgrp [-] [groupid]
-	 *      newgrp [-l] [groupid]
-	 *      sg [-] groupid [[-c] command]
-	 *      sg [-l] groupid [[-c] command]
-	 */
-	if (   (argc > 0)
-	    && (   streq(argv[0], "-")
-	        || streq(argv[0], "-l"))) {
-		argc--;
+	if (*argv != NULL && (streq(*argv, "-l") || streq(*argv, "-"))) {
 		argv++;
 		initflag = true;
 	}
-	if (!is_newgrp) {
-		/*
-		 * Do the command line for everything that is
-		 * not "newgrp".
-		 */
-		if ((argc > 0) && (argv[0][0] != '-')) {
-			if (!is_valid_group_name (argv[0])) {
-				fprintf (
-					stderr, _("%s: provided group is not a valid group name\n"),
-					Prog);
-				goto failure;
-			}
-			group = argv[0];
-			argc--;
-			argv++;
-		} else {
-			usage ();
-			closelog ();
-			exit (EXIT_FAILURE);
-		}
-		if (argc > 0) {
-
-			/*
-			 * Skip -c if specified so both forms work:
-			 * "sg group -c command" or "sg group command".
-			 */
-			if ((argc > 1) && streq(argv[0], "-c")) {
-				command = argv[1];
-			} else {
-				command = argv[0];
-			}
-			cflag = true;
+	if (*argv != NULL && strspn(*argv, "-")) {
+		usage();
+		goto failure;
+	}
```

Comment:
This whole block can be removed, since this check is already covered in `is_valid_group_name`.